libraries/chain/fork_database.cpp

```diff
@@ -128,7 +128,7 @@ namespace eosio { namespace chain {
                for( const auto& receipt : s.block->transactions ) {
```

Comment:
We should probably skip this whole loop and temporarily remove the `trxs` field because we don't actually use it anyway. We really need a separate PR to more sensibly handle caching transaction metadatas that are created as part of applying blocks. Currently the fork database will only cache `transaction_metadata`s for blocks produced by a local producer.

Reply:
I'm going to go ahead and clean this up on this PR.

Reply:
Reused cached `transaction_metadata` if available and created `completed_block` with existing `block_state_ptr` in `apply_block`. This also has the benefit of not having to create a new `block_state`.